### PR TITLE
Include draft flag

### PR DESCRIPTION
**Pull request by Kastet**
Adds `--includeDraft` flag to the test runner CLI that allows to include draft endpoint tests during development. Documentation not included.

### CODE REVIEW

File: lib/src/testing/test-runner.spec.ts [hunk 1]
```diff
@@ -176,6 +176,26 @@ describe("test runner", () => {
...
+      .post("/companies", { name: "My Company", private: true })
+      .reply(201, { name: "My Company" })
+      .post("/state/teardown")
+      .reply(200);
+
+    const result = await testRunner.test(contract, {
+      includeDraft: true
+    });
+    expect(scopeDraft.isDone()).toBe(true);
+    expect(result).toBe(true);
```
Comment (fwouts) [inline lib/src/testing/test-runner.spec.ts:196]: Would it be worth having another test where it fails, and confirming that it's true when includeDraft is false, and false when includeDraft is true?

Comment (Kastet) [reply lib/src/testing/test-runner.spec.ts]: The check here is line 195 - `expect(scopeDraft.isDone()).toBe(true);`.
There is an [existing test](https://github.com/airtasker/spot/blob/test-runner-include-draft-flag/lib/src/testing/test-runner.spec.ts#L150) that checks that without `includeDraft` draft tests are excluded `expect(scopeDraft.isDone()).toBe(false);`

Comment (Kastet) [reply lib/src/testing/test-runner.spec.ts:196]: Added

File: lib/src/testing/test-runner.spec.ts [hunk 2]
```diff
@@ -176,7 +176,7 @@ describe("test runner", () => {
     expect(result).toBe(true);
   });
 
-  test("include draft", async () => {
+  test("draft endpoint included", async () => {
     const contract = parseAndCleanse(
       `${testExamplesBasePath}/draft-endpoint.ts`
```
Comment (fwouts) [inline lib/src/testing/test-runner.spec.ts:181]: Is there a reason why the contract used isn't the same as for `draft endpoint excluded`?

File: lib/src/testing/test-runner.spec.ts [hunk 3]
```diff
@@ -196,6 +196,46 @@ describe("test runner", () => {
     expect(result).toBe(true);
   });
 
+  test("draft endpoint included - response body mismatch", async () => {
```
Comment (fwouts) [inline lib/src/testing/test-runner.spec.ts]: To be clear, what I'd like the test to be responsible for is `failing draft endpoint fails the test when draft endpoints are included`

(I'm not entirely sure the test below is necessary)

**Review by fwouts — APPROVED** [review]

**Pull request merged**